libs/ccf-shared/src/lib/services/data-source/api-endpoint.service.ts

```diff
@@ -219,4 +203,28 @@ export class ApiEndpointDataSourceService implements DataSource {
       map((data) => (reviver ? reviver(data) : data)),
     );
   }
+
+  private checkForSources(): Observable<unknown> {
+    return combineLatest([this.globalConfig.getOption('dataSources'), this.globalConfig.getOption('filter')]).pipe(
+      debounceTime(1000),
+      tap(([sources, filter]) => {
+        if ((sources && sources.length > 0) || filter) {
+          const sessionTokenRequest = {
+            dataSources: sources,
+            filter: filter,
+          };
+          this.getSessionToken({ sessionTokenRequest } as SessionTokenRequestParams);
+        }
+      }),
+    );
+  }
+
+  private getSessionToken(params: SessionTokenRequestParams) {
+    this.api.sessionToken(params).subscribe((resp: DefaultParams) => {
+      const token = resp.token;
+      this.globalConfig.patchState({ token });
+      localStorage.setItem('SESSION_TOKEN', token ?? '');
```

Comment:
Should we really save the token in localStorage? Do we ever read it from localStorage?

Reply:
No we don't need to do that anymore